reps/2023-10-30-gpu-memory-support.md

```diff
@@ -0,0 +1,266 @@
+# Ray GPU Memory Support
+
+## Summary
+
+Design docummentation for supporting gpu-memory on ray core resource scheduler.
+
+### General Motivation
+
+Currently, `ray` supports `num_gpus` to scheduler resource to tasks/actors, which then assign either `num_gpus` number of gpu ids to be used by the tasks/actors. Additionally, `ray` provides support fractional gpus allocation by specifying `num_gpus < 1` so a single GPU can be used to run multiple tasks. This works well if the cluster only has a single type of GPUs. However, imagining a cluster has both A100 40GB and A100 80GB GPUs, setting num_gpus to a fixed number doesn’t work that well: if we set to 0.1 then we will get 4GB if the scheduler picks A100 40GB but 8GB if the scheduler picks A100 80GB which is a waste of resource. We can also set accelerator_type to A100_40GB and num_gpus to 0.1 to make sure we get the exact amount of GPU memory we need but then the task cannot run on A100 80GB even if it’s free.
+
```

Comment:
We should move user issues up here.

You can say that many Ray users also encountered this limitations: links

Reply:
ohh I put this under `Ray community users’s demand`

Reply:
Yea, I'm suggesting we move that up to here so problems and then solution.